features/steps/ruby_notifier_steps.rb

```diff
@@ -1,31 +1,79 @@
+    And the payload field "notifier.version" is not null
+    And the payload field "events" is a non-empty array
+
+    And each element in payload field "events" has "severity"
+    And each element in payload field "events" has "severityReason.type"
+    And each element in payload field "events" has "unhandled"
+    And each element in payload field "events" has "exceptions"
   }
 end
 
-When("I set environment variable {string} to the proxy settings with credentials {string}") do |env_var, credentials|
+Given("I start the rails service") do
+  rails_version = ENV["RAILS_VERSION"]
   steps %Q{
-    When I set environment variable "#{env_var}" to "#{credentials}@#{current_ip}:#{MOCK_API_PORT}"
+    When I start the service "rails#{rails_version}"
+    And I wait for the host "rails#{rails_version}" to open port "3000"
   }
 end
 
-Then(/^the "(.+)" of the top non-bugsnag stackframe equals (\d+|".+")(?: for request (\d+))?$/) do |element, value, request_index|
-  stacktrace = read_key_path(find_request(request_index)[:body], 'events.0.exceptions.0.stacktrace')
-  frame_index = stacktrace.find_index { |frame| ! /.*lib\/bugsnag.*\.rb/.match(frame["file"]) }
+When("I navigate to the route {string} on the rails app") do |route|
+  rails_version = ENV["RAILS_VERSION"]
   steps %Q{
-    the "#{element}" of stack frame #{frame_index} equals #{value}
+    When I open the URL "http://rails#{rails_version}:3000#{route}"
   }
 end
 
-Then(/^the total sessionStarted count equals (\d+)(?: for request (\d+))?$/) do |value, request_index|
-  session_counts = read_key_path(find_request(request_index)[:body], "sessionCounts")
-  total_count = session_counts.inject(0) { |count, session| count += session["sessionsStarted"] }
-  assert_equal(value, total_count)
+Then("the payload field {string} matches the appropriate handled JSON fixture") do |field|
+  if ENV["SIDEKIQ_VERSION"] == "~> 2"
+    created_at_present = "false"
+  else
+    created_at_present = "true"
+  end
+  steps %Q{
+    And the payload field "#{field}" matches the JSON fixture in "features/fixtures/sidekiq/payloads/handled_metadata_ca_#{created_at_present}.json"
+  }
+end
+
+Then("the payload field {string} matches the appropriate unhandled JSON fixture") do |field|
+  if ENV["SIDEKIQ_VERSION"] == "~> 2"
+    created_at_present = "false"
+  else
+    created_at_present = "true"
+  end
+  steps %Q{
+    And the payload field "#{field}" matches the JSON fixture in "features/fixtures/sidekiq/payloads/unhandled_metadata_ca_#{created_at_present}.json"
+  }
+end
+
+Then("the event contains a breadcrumb matching the JSON fixture in {string}") do |json_fixture|
+  breadcrumbs = read_key_path(Server.current_request[:body], "events.0.breadcrumbs")
+  expected = JSON.parse(open(json_fixture, &:read))
+  match = breadcrumbs.any? { |breadcrumb| value_compare(expected, breadcrumb).equal? }
+  assert(match, "No breadcrumbs in the event matched the given breadcrumb")
```

Comment:
Should this be moved to Maze Runner?

Reply:
PR present [here](https://github.com/bugsnag/maze-runner/pull/73)